arduino/waveform/Source.h

```diff
@@ -0,0 +1,161 @@
+/** @file Source.h
+  
+    @author Ananya Bahadur
+*/
+
+#ifndef CL_SOURCE_H
+#define CL_SOURCE_H
+
+#if ARDUINO >= 100
+ #include <Arduino.h>
+#else
+ #include <WProgram.h>
+#endif
+
+#include "Cyclops.h"
+
+/** @typedef operationMode */
+typedef enum {
+    LOOPBACK,
+    ONE_SHOT,
+} operationMode;
+
+/** @typedef sourceStatus */
+typedef enum {
+    FROZEN,
+    ACTIVE
+} sourceStatus;
+
+#define ONE_SHOT_FINISHED_HOLD_TIME 1000 /**< The waveform is completed.  
```

Comment:
This was there only for testing and **will go away**. It does kind-of hardcode the time between one shot waveforms, but that was not the intention.
The CPU is busy-waiting to process the one-shot waveform, and compute the next "waiting-period". When the waveform is exhausted, the CPU need not wait for any more waveform processing. But I let it wait for an update once every 1000 units, for simplicity in test implementation. Nothing happens in these updates.
The scheduler will handle all this correctly. It would be able to view waveform state and decide whether any processing needs to be done.